src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.ValueTaskSource.cs

```diff
@@ -200,6 +170,9 @@ private void CompleteCallback(ulong packedResult)
                     Debug.Assert(result == TaskSourceCodes.ResultSuccess, "Unknown result");
                     _source.SetResult((int)(packedResult & uint.MaxValue));
                 }
+
+                // The instance is ready to be reused
+                _strategy.TryToReuse(this);
```

Comment:
This is wrong and must be fixed before this is merged.  This can't be stored for reuse until GetResult is called on VTS.